components/BottomNavigation/src/private/MDCBottomNavigationItemView.m

```diff
@@ -0,0 +1,229 @@
+      self.label.hidden = NO;
+    } else if (self.titleHideState == MDCBottomNavigationBarTitleHideStateAlways) {
+      self.label.hidden = YES;
+      iconImageViewCenter = CGPointMake(CGRectGetMidX(self.bounds), CGRectGetMidY(self.bounds));
+    }
+    self.label.textColor = self.selectedItemTintColor;
+    self.iconImageView.image = self.selectedImage;
+  } else {
+    if (self.titleHideState == MDCBottomNavigationBarTitleHideStateDefault ||
+        self.titleHideState == MDCBottomNavigationBarTitleHideStateAlways) {
+      self.label.hidden = YES;
+      iconImageViewCenter = CGPointMake(CGRectGetMidX(self.bounds), CGRectGetMidY(self.bounds));
+    } else if (self.titleHideState == MDCBottomNavigationBarTitleHideStateNever) {
+      self.label.hidden = NO;
+    }
+    self.label.textColor = self.unselectedItemTintColor;
+    self.iconImageView.image = self.unselectedImage;
+  }
+  CGPoint badgeCenter =
+      CGPointMake(CGRectGetMidX(self.bounds) + CGRectGetWidth(self.iconImageView.bounds) / 2,
+                  iconImageViewCenter.y - CGRectGetMidX(self.iconImageView.bounds));
+  self.label.center =
+      CGPointMake(CGRectGetMidX(self.bounds),
+                  CGRectGetMidY(self.bounds) + CGRectGetHeight(self.bounds) * 0.25f);
+  if (animated) {
+    [UIView animateWithDuration:kMDCBottomNavigationItemViewTransitionDuration animations:^(void) {
+      self.iconImageView.center = iconImageViewCenter;
+      self.badge.center = badgeCenter;
+    }];
+  } else {
+    self.iconImageView.center = iconImageViewCenter;
+    self.badge.center = badgeCenter;
+  }
+}
+
+- (void)setInterimSelected:(BOOL)interimSelected {
+  if (interimSelected) {
+    self.circleLayer.opacity = kMDCBottomNavigationItemViewCircleOpacity;
+    self.iconImageView.image = self.selectedImage;
+  } else {
+    self.circleLayer.opacity = 0;
+    self.iconImageView.image = self.unselectedImage;
+  }
+}
+
+- (void)setSelectedItemTintColor:(UIColor *)selectedItemTintColor {
+  _selectedItemTintColor = selectedItemTintColor;
+  self.label.textColor = self.selectedItemTintColor;
+  self.selectedImage = [self colorizeImage:self.image color:self.selectedItemTintColor];
+  self.circleLayer.fillColor = self.selectedItemTintColor.CGColor;
+}
+
+- (void)setUnselectedItemTintColor:(UIColor *)unselectedItemTintColor {
+  _unselectedItemTintColor = unselectedItemTintColor;
+  self.unselectedImage = [self colorizeImage:self.image color:self.unselectedItemTintColor];
+}
+
+- (void)setBadgeColor:(UIColor *)badgeColor {
+  _badgeColor = badgeColor;
+  self.badge.badgeColor = badgeColor;
+}
+
+- (void)setBadgeValue:(NSString *)badgeValue {
+  _badgeValue = badgeValue;
+  self.badge.badgeValue = badgeValue;
+  if (!_badgeValue) {
+    self.badge.hidden = YES;
+  } else {
+    self.badge.hidden = NO;
+  }
+}
+
+- (void)setImage:(UIImage *)image {
+  _image = image;
+
+  self.selectedImage = [self colorizeImage:image color:self.selectedItemTintColor];
```

Comment:
Rather than storing two images I would expect us to use UIImageView.tintColor for this.
self.iconImageView.image = image;
self.iconImageView.tintColor = self.unselectedTintColor;

On setSelected we would update iconImageView.tintColor.

Reply:
Done. Now using tintColor. Needed to use imageWithRenderingMode.